README.md

```diff
@@ -1,46 +1,66 @@
-# Advanced Sample Hardhat Project
-
-This project demonstrates an advanced Hardhat use case, integrating other tools commonly used alongside Hardhat in the ecosystem.
-
-The project comes with a sample contract, a test for that contract, a sample script that deploys that contract, and an example of a task implementation, which simply lists the available accounts. It also comes with a variety of other tools, preconfigured to work with the project code.
-
-Try running some of the following tasks:
-
-```shell
-npx hardhat accounts
-npx hardhat compile
-npx hardhat clean
-npx hardhat test
-npx hardhat node
-npx hardhat help
-REPORT_GAS=true npx hardhat test
-npx hardhat coverage
-npx hardhat run scripts/deploy.ts
-TS_NODE_FILES=true npx ts-node scripts/deploy.ts
-npx eslint '**/*.{js,ts}'
-npx eslint '**/*.{js,ts}' --fix
-npx prettier '**/*.{json,sol,md}' --check
-npx prettier '**/*.{json,sol,md}' --write
-npx solhint 'contracts/**/*.sol'
-npx solhint 'contracts/**/*.sol' --fix
+# About
+The Typescript library for developers to use with Medusa-enabled applications
+
+# Installation
+
+```bash
+npm install --save @medusa-network/medusa-sdk
 ```
 
-# Etherscan verification
+# Usage
 
-To try out Etherscan verification, you first need to deploy a contract to an Ethereum network that's supported by Etherscan, such as Ropsten.
+[See the demo application for more examples of usage](https://github.com/medusa-network/medusa-app)
 
-In this project, copy the .env.example file to a file named .env, and then edit it to fill in the details. Enter your Etherscan API key, your Ropsten node URL (eg from Alchemy), and the private key of the account which will send the deployment transaction. With a valid .env file in place, first deploy your contract:
+```typescript
+import { Medusa, EVMG1Point, SuiteType } from "@medusa-network/medusa-sdk";
 
-```shell
-hardhat run --network ropsten scripts/deploy.ts
-```
+// Initialize medusa with using the bn254 curve with the keypair in G1
+// using the HGamal encryption algorithm
+const medusa = await Medusa.init(SuiteType.BN254_KEYG1_HGAMAL);
+
+// Get Public Key of Medusa Oracle contract
+// Note: Medusa can be initialized one or more times for a given encryption suite
+const medusa = await Medusa.init(SuiteType.BN254_KEYG1_HGAMAL);
+const medusaContract = new ethers.Contract(
+  "0x..",
+  ["function distributedKey() external view returns (G1Point memory)"],
+  signerOrProvider
+);
+const medusaPublicKey = medusa.decodePublicKey(
+  await medusaContract.distributedKey()
+);
+
+// Encrypt data towards Medusa
+const { encryptedData, encryptedKey } = await medusa.encrypt(
+  "This is secret!",
+  medusaPublicKey,
+  myApplicationContractAddress,
+  userAddress
+);
 
-Then, copy the deployment address and paste it in to replace `DEPLOYED_CONTRACT_ADDRESS` in this command:
+// Derive a user's keypair from their (deterministic) signature
+const keypair = medusa.calculateKeypair(signature);
```

Comment:
`deriveKeyPair` sounds indeed a better name. And again, can we integrate with web3 provider or stg to automatically ask the signature ?

Reply:
Oh yes need to update to use the new name `deriveKeypair`.

For the signature/key, I'll see what makes sense. Maybe I can store the derivedKey as a global on the window so the user does not have to sign more times than necessary (if Medusa.deriveKeypair is called on different pages for example).

I think this approach is safe from a security standpoint, but let me double check.